### PR TITLE
Feat/react time slots components

### PR DESCRIPTION
**Pull request by tmkuba**
Did `<TimeSlot />` and `<TimeSlotSelector />` Components together
https://trello.com/c/vqkvbOKm
https://trello.com/c/BD6yXItO

### CODE REVIEW

File: client/src/components/TimeSlot.jsx [hunk 1]
```diff
@@ -1,5 +1,22 @@
...
 
-const TimeSlot = () => <div>TimeSlot</div>;
+const TimeSlot = props => (
+  <div>
+    <button
+      disabled={!props.available}
+      onClick={() => props.clickHandler(props.time)}
+    >
+      {props.time > 12 ? props.time - 12 : props.time}:00 {props.time > 12 ? 'PM' : 'AM'}
+    </button>
```
Comment (matgbx) [inline client/src/components/TimeSlot.jsx:11]: nice,  I like this ternary -  very simple one liner

File: client/src/components/TimeSlotSelector.test.jsx [hunk 2]
```diff
@@ -2,9 +2,141 @@ import React from 'react';
...
+      bookingsMadeToday={25}
+      availabilityInfo={[
+        { time: 17, remaining: 0 },
+        { time: 18, remaining: 5 },
+        { time: 19, remaining: 3 },
+        { time: 20, remaining: 2 },
+        { time: 21, remaining: 1 },
+      ]}
+      clickHandler={() => {}}
+    />)).toMatchSnapshot();
```
Comment (matgbx) [inline client/src/components/TimeSlotSelector.test.jsx:20]: I'm guessing you don't need to pass the actual function through here as it is only comparing the snapshot?

**Review by matgbx — APPROVED** [review]
Looks all good to me

File: client/src/components/TimeSlotSelector.test.jsx (hunk 2 above)
Comment (tmkuba) [reply client/src/components/TimeSlotSelector.test.jsx:20]: yeah, TimeSlotSelector only passes its clickHandler down to each child TimeSlot component and never actually invokes it itself.  The TimeSlot component test makes sure the clickHandler is called properly.

**Review by leungnamson1023 — COMMENTED** [review]
looks good!

**Pull request merged**